src/ffi.rs

```diff
@@ -416,20 +412,16 @@ pub fn nullifiers(args: i32, len: i32) -> i64 {
     };
 
     let mut nullifiers = Vec::with_capacity(notes.len());
-    let mut sks = unsafe { [mem::zeroed(); MAX_KEY + 1] };
-    let mut vks = unsafe { [mem::zeroed(); MAX_KEY + 1] };
-    let mut keys_len = 0;
+
+    let sks: [SecretKey; MAX_KEY] =
+        core::array::from_fn(|i| key::derive_sk(&seed, i as _));
+    let vks: [ViewKey; MAX_KEY] =
+        core::array::from_fn(|i| key::derive_vk(&seed, i as _));
 
     'outer: for note in notes {
```

Comment:
You still didn't modify this loop as discussed.

1. The comment is wrong, or best case scenario misleading
2. The usage of labels makes the flow harder to follow from top to bottom – it's "GOTO-like" –; therefore the labels should be used only when there is not any decent alternative available
3. Due to the label usage, the code makes use of a temp var called `keys_len` that shouldn't be there.

All those should be fixed, if you don't want to do it in this PR feel free to open immediately a new one – but also an issue related to that – to apply those changes. I suggested to do it in this PR since you're already touching this code and in addition you can skip to create a separate issue. But if you prefer to file a new issue and fix it immediately in a new PR, it's also fine.

My suggestion, as I already mentioned to you, would be modify this using a more functional approach with `try_fold`. Here is the code:

```rust
    let nullifiers = notes.iter().try_fold(
        Vec::with_capacity(notes.len()),
        |mut nullifiers, note| {
            if let Some(idx) = (0..MAX_KEY).find(|&i| vks[i].owns(note)) {
                let nullifier = note.gen_nullifier(&sks[idx]);
                nullifiers.push(nullifier);
                Ok(nullifiers)
            } else {
                Err(())
            }
        },
    );

    match nullifiers {
        Ok(n) => utils::rkyv_into_ptr(n),
        Err(_) => utils::fail(),
    }
```

Basically, if you cannot decrypt a note with *any* of the keys you own, than the function calls `utils::fail`.

Reply:
I removed the comment and the label, I think any greater refractor than that would be counter productive since the "big" refractor is imminent, I have already created the EPIC for refractor I'll add this content to it 
https://github.com/dusk-network/wallet-core/issues/117